### PR TITLE
Update to datashader 0.15.0

### PR DESCRIPTION
**Pull request by philippjfr**
Changelog: https://github.com/holoviz/datashader/blob/v0.15.0/CHANGELOG.rst
License: https://github.com/holoviz/datashader/blob/v0.15.0/LICENSE.txt
Requirements:
- https://github.com/holoviz/datashader/blob/v0.15.0/pyproject.toml
- https://github.com/holoviz/datashader/blob/v0.15.0/setup.py

Changes:
- Removes obsolete patches
- Resets build number
- Updates version and SHA

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,9 +19,6 @@ build:
     - datashader = datashader.__main__:main
 
 requirements:
-  build:
-    - m2-patch  # [win]
-    - patch     # [not win]
   host:
     - python
     - pip
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:24]: `numba >=0.51` should be fixed as `numba`, https://github.com/holoviz/datashader/blob/28c85812abf1ac55766e71e36677d4d5e55b6ff5/setup.py#L16

**Review by skupr-anaconda — APPROVED** [review]
LGTM but a small fix

File: recipe/meta.yaml (hunk 1 above)
Comment (philippjfr) [reply recipe/meta.yaml:24]: Done.

**Review by akabanovs — APPROVED** [review]
LGTM

**Review by ianthomas23 — APPROVED** [review]

**Pull request merged**